core/executor/src/wasm_executor.rs

```diff
@@ -1371,43 +1386,32 @@ impl WasmExecutor {
 			},
 		};
 
-		// cleanup module instance for next use
-		let new_low = memory.lowest_used();
-		if new_low < low {
-			memory.zero(new_low as usize, (low - new_low) as usize)?;
-			memory.reset_lowest_used(low);
-		}
-		memory.with_direct_access_mut(|buf| buf.resize(used_mem.0, 0));
 		result
 	}
 
 	/// Prepare module instance
-	pub fn prepare_module<E: Externalities<Blake2Hasher>>(
-		&self,
-		ext: &mut E,
+	pub fn instantiate_module<E: Externalities<Blake2Hasher>>(
 		heap_pages: usize,
 		module: &Module,
-		) -> Result<ModuleRef>
-	{
+	) -> Result<ModuleRef> {
 		// start module instantiation. Don't run 'start' function yet.
 		let intermediate_instance = ModuleInstance::new(
 			module,
 			&ImportsBuilder::new()
 			.with_resolver("env", FunctionExecutor::<E>::resolver())
 		)?;
 
-		// extract a reference to a linear memory, optional reference to a table
-		// and then initialize FunctionExecutor.
+		// extract a reference to a linear memory and extract the `__heap_base` start
 		let memory = Self::get_mem_instance(intermediate_instance.not_started_instance())?;
+		let _heap_base = Self::get_heap_base(intermediate_instance.not_started_instance())?;
```

Comment:
Is the purpose of this just to verify that the heap_base is set? If so the comment should reflect this -- the term "extract" implies to me that it will be used.

Reply:
Ahh, you are right! Yes, we check that `__heap_base` exists (and return `Err` if it doesn't`). I was also thinking about leaving the comment but apparently forgot about it!